frontend/packages/core/src/Network/errors.ts

```diff
@@ -0,0 +1,290 @@
+   */
+  resourceName?: string;
+  /**
+   * The owner of the resource (optional).
+   *
+   * For example, "user:<owner email>" or "project:<Google developer project id>".
+   */
+  owner?: string;
+  /**
+   * Describes what error is encountered when accessing this resource.
+   *
+   * For example, updating a cloud project may require the `writer` permission
+   * on the developer console project.
+   */
+  description?: string;
+}
+
+/**
+ * Provides links to documentation or for performing an out of band action.
+ *
+ * For example, if a quota check failed with an error indicating the calling
+ * project hasn't enabled the accessed service, this can contain a URL pointing
+ * directly to the right place in the developer console to flip the bit.
+ */
+export interface Help {
+  /** URL(s) pointing to additional information on handling the current error. */
+  links?: {
+    /** Describes what the link offers. */
+    description?: string;
+    /**
+     * The URL of the link.
+     */
+    url?: string;
+  }[];
+}
+
+/** An error received from the backend of Clutch. */
+export interface ClutchError extends Error {
+  /** The HTTP status information. */
+  status: HttpStatus;
+  /** The status code. */
+  code?: number;
+  /**
+   * A developer-facing error message in English. Any user-facing error messages
+   * will be sent in the details field.
+   */
+  message: string;
+  /** A list of objects that carry error details from the server. */
+  details?: (
+    | RetryInfo
+    | DebugInfo
+    | QuotaFailure
+    | ErrorInfo
+    | PreconditionFailure
+    | BadRequest
+    | RequestInfo
+    | ResourceInfo
+    | Help
+    | any
+  )[];
+  /** Data present on the response object, if any. */
+  data?: any;
+}
+
+const GRPC_DETAIL_TYPES_MAP = {
+  "types.google.com/google.rpc.RetryInfo": {} as RetryInfo,
+  "types.google.com/google.rpc.DebugInfo": {} as DebugInfo,
+  "types.google.com/google.rpc.QuotaFailure": {} as QuotaFailure,
+  "types.google.com/google.rpc.ErrorInfo": {} as ErrorInfo,
+  "types.google.com/google.rpc.PreconditionFailure": {} as PreconditionFailure,
+  "types.google.com/google.rpc.BadRequest": {} as BadRequest,
+  "types.google.com/google.rpc.RequestInfo": {} as RequestInfo,
+  "types.google.com/google.rpc.ResourceInfo": {} as ResourceInfo,
+  "types.google.com/google.rpc.Help": {} as Help,
+};
+
+/**
+ * Construct a ClutchError from an AxiosError.
+ *
+ * @param clientError A client error object.
+ */
+const grpcResponseToError = (clientError: AxiosError): ClutchError => {
+  const { data } = clientError?.response;
+
+  const error = {
+    code: data.code,
+    message: data.message,
+    status: {
+      code: clientError.response.status,
+      text: clientError.response.statusText,
+    } as HttpStatus,
+  } as ClutchError;
+
+  if (data?.details !== undefined && data.details.length > 0) {
+    let details = GRPC_DETAIL_TYPES_MAP[data.details["@type"]];
```

Comment:
details is always a list, so i'm not clear how `details["@type"]` would work, doesn't that need to be an array index?

Reply:
oh good catch. This should be a a check on the individual details.

Reply:
I removed this all together since I don't think it adds value here.